[DOC-664][2.25.1] PG15 upgrade using yugabyted (#26373)
yugabyte/yugabyte-db: docs/content/preview/reference/configuration/yugabyted.md

### finalize_upgrade

> {{< warning title="Upgrading to v2.25" >}}

Reviewer: @nchandrappa this isn't specific to 2.25 though, all upgrades will use the new way right?

Author: Yes, that is correct. All new upgrades going forward need to follow these steps.

--base_dir *base-directory*
: The base directory for the yugabyted server that needs to be stopped.

--upgrade *stop-for-upgrade*
:  Stop the yugabyted node for version upgrade.

-----

### upgrade

> Use the `yugabyted upgrade` to prepare and perform YugabyteDB major version upgrade. 

Author: From going forward, v2.25.1 onwards all upgrades need to follow these steps. It will be the same with `v2025.1` when it gets released.

--base_dir *base-directory*
: The base directory for the yugabyted server that needs to be stopped.

--upgrade *bool*
:  Stop the yugabyted node for version upgrade. Default: `false`.

-----

### upgrade

Use the `yugabyted upgrade` to prepare and perform YugabyteDB major version upgrade. 

```sh
Usage: yugabyted upgrade [command] [flags]
```

#### Commands

The following sub-commands are available for `yugabyted upgrade` command:

- [ysql_catalog](#ysql-catalog)
- [finalize_new_version](#finalize-new-version)
- [check_version_compatibility](#check-version-compatibility)

#### ysql_catalog

Use the sub-command `yugabyted upgrade ysql_catalog` to upgrade the ysql catalog of the YugabyteDB cluster to a newer version. This command needs
to be executed only once per upgrade from a node that is part of the cluster.

> For example to upgrade the ysql catalog of a cluster, user will need to execute the below command:

Reviewer: ```suggestion
For example, to upgrade the YSQL catalog of a cluster, you would execute the following command:
```

```sh
./bin/yugabyted upgrade ysql_catalog
```

##### ysql_catalog flags

-h | --help
: Print the command-line help and exit.

--base_dir *base-directory*
: The base directory for the yugabyted server.

--timeout *timeout*
: Custom timeout for the YSQL catalog upgrade in milliseconds.

#### finalize_new_version

Use the sub-command `yugabyted upgrade finalize_new_version` to finalize the new version on the upgraded YugabyteDB Cluster. This command needs
to be executed only once per upgrade from a node that is part of the cluster.

> For example to finalize the new version on a cluster, user will need to execute the below command:

Reviewer: ```suggestion
For example, to finalize an upgrade to a cluster, you would execute the following command:
```

```sh
./bin/yugabyted upgrade finalize_new_version
```

##### finalize_new_version flags

-h | --help
: Print the command-line help and exit.

--base_dir *base-directory*
: The base directory for the yugabyted server.

--timeout *timeout*
: Custom timeout for the YSQL catalog upgrade in milliseconds.

#### check_version_compatibility

Use the sub-command `yugabyted upgrade check_version_compatibility` to verify if the exisitng YugabyteDB cluster is compatible upgrade to the new version. 
This command needs to be executed only once per upgrade from a node that is part of the cluster.

> For example to verify the version compatibility of the cluster to a new version, user will need to execute the below command:

Reviewer: ```suggestion
For example, to verify the compatibility of a cluster with a new version, you would execute the following command:
```

> ## Upgrade a YugabyteDB cluster

Reviewer 2: Remove this entire section. This is not covered in the regular upgrade section

Reviewer 2: Actually lets move it to the normal upgrade doc page.

1. Start the new yugabyted process (from the new downloaded release) by executing the `yugabyted start` command. Use the previously configured `--base_dir` when restarting the instance.

    ```sh
    ./bin/yugabyted start --base_dir <path_to_base_dir>
    ```

1. Repeat steps 2 and 3 for all the nodes. Wait 60 seconds before repeating the steps on each node.

1. After restarting all the nodes, upgrade the YSQL catalog of the cluster:

    ```sh
    ./bin/yugabyted upgrade ysql_catalog --base_dir <path_to_base_dir>
    ```

1. We need to restart all the nodes for the second time after successful ysql catalog upgrade. 

    Repeat steps 2 and 3 for all the nodes. Need to wait 60 seconds before repeating the steps on each node.

1. After restarting all the nodes. Finalize the upgrade by running `yugabyted finalize_new_version` command. This command can be run from any node.

    ```sh
    ./bin/yugabyted upgrade finalize_new_version --base_dir <path_to_base_dir>
    ```

    Use the `timeout` flag to specify custom upgrade timeout. Default value is 60000 ms.

    ```sh
    ./bin/yugabyted upgrade finalize_new_version --base_dir <path_to_base_dir> --timeout 80000
    ```

> ### Upgrade a cluster from single to multi zone

Reviewer: @nchandrappa do these steps not need to use the new commands

## Upgrade a YugabyteDB cluster

{{< warning title="Upgrading to v2.25" >}}
Upgrading to v2.25.1 is only supported from v2024.2.2 release.
{{< /warning >}}

To use the latest features of the database and apply the latest security fixes, upgrade your YugabyteDB cluster to the [latest release](https://download.yugabyte.com/#/).

Upgrading an existing YugabyteDB cluster that was deployed using yugabyted includes the following steps:

1. Verify the version compatibility before upgrading the cluster.

    ```sh
    ./bin/yugabyted upgrade check_version_compatibility

> 

Reviewer: ```suggestion
```

    ```

1. Stop one of the running YugabyteDB node using the `yugabyted stop` command.

    ```sh
    ./bin/yugabyted stop --upgrade true --base_dir <path_to_base_dir>
    ```

    Note: Wait for 60 seconds before restarting the node.

1. Start the new yugabyted process (from the new downloaded release) by executing the `yugabyted start` command. Use the previously configured `--base_dir` when restarting the instance.

    ```sh
    ./bin/yugabyted start --base_dir <path_to_base_dir>
    ```

> 1. Repeat steps 2 and 3 for all the nodes. Need to wait 60 seconds before repeating the steps on each node.

Reviewer: ```suggestion
1. Repeat steps 2 and 3 for all the nodes. Wait 60 seconds before repeating the steps on each node.
```

> 1. After restarting all the nodes. Upgrade YSQL catalog of the YugabyteDB cluster using the below command:

Reviewer: ```suggestion
1. After restarting all the nodes, upgrade the YSQL catalog of the cluster:
```
